src/Groups/iso_oscar_gap.jl

```diff
@@ -0,0 +1,103 @@
+import AbstractAlgebra: MatElem
+import Hecke: FqNmodFiniteField, FqNmodMatSpace, fq_nmod, fq_nmod_mat
+import GAP: FFE
+
+# isomorphism from the Oscar field to the GAP field
+struct GenRingIso <: Map{FqNmodFiniteField,GapObj,SetMap,GenRingIso}
+   domain::FqNmodFiniteField
+   codomain::GapObj
+   f
+   f_inv
+end
+
+# turns the Oscar matrix into the GAP matrix and viceversa
+struct GenMatIso <: Map{FqNmodMatSpace,GapObj,SetMap,GenMatIso}
+   domain::FqNmodMatSpace
+   codomain::GapObj
+   fr::GenRingIso
+   f
+   f_inv
+end
+
+
+################################################################################
+#
+#  Ring isomorphism
+#
+################################################################################
+
+function elem_f(x::fq_nmod; B=0,d=0)
+   COEF = [Int64(coeff(x,i)) for i in 0:d-1]
+   x_gap = sum([COEF[i]*B[i] for i in 1:d])
+   return x_gap
+end
+
+function elem_g(x::FFE; B=0, z=0, d=0)
+   L = GAP.gap_to_julia(GAP.Globals.List(GAP.Globals.Coefficients(B,x),GAP.Globals.IntFFE ))
+   return sum([L[i]*z^(i-1) for i in 1:d])
+end
+
+function gen_ring_iso(F::FqNmodFiniteField)
+   p = Int64(characteristic(F))
+   d = Int64(degree(F))
+   z = gen(F)
+
+   if d==1
+      f(x::fq_nmod) = Int(coeff(x,0))*GAP.Globals.One(GAP.Globals.GF(p))
```

Comment:
This will call `GAP.Globals.One(GAP.Globals.GF(p))` each time it is called. Better to store that value and use it in a closure. Here's some micro benchmark:
```julia
function HACK(p::Int)
  tmpval = GAP.Globals.One(GAP.Globals.GF(p))
  f(x) = x*GAP.Globals.One(GAP.Globals.GF(p))
  g(x) = x* tmpval
  return f, g
end
```
and then
```
julia> using BenchmarkTools

julia> f31, g31 = HACK(31)
(var"#f#3"{Int64}(31), var"#g#4"{FFE}(GAP: Z(31)^0))

julia> @btime f31(1)
  426.919 ns (5 allocations: 96 bytes)
GAP: Z(31)^0

julia> @btime g31(1)
  100.483 ns (2 allocations: 32 bytes)
GAP: Z(31)^0

```
So this is a factor 4 in performance in this example.

Reply:
This is still not resolved. Can you perhaps just try the following:
```suggestion
      myOne = GAP.Globals.One(GAP.Globals.GF(p))
      f(x::fq_nmod) = Int(coeff(x,0))*myOne
```
I.e., does that produce correct results? Can you compare it with the existing code and using `BenchmarkTools`  / `@btime` as I did above?

Reply:
Done it. Running time and allocated memory get halved.